pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -0,0 +1,511 @@
+	}
+	if egressNPName, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyName"); ok {
+		chFlowRow.egressNetworkPolicyName = egressNPName.GetStringValue()
+	}
+	if egressNPNamespace, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyNamespace"); ok {
+		chFlowRow.egressNetworkPolicyNamespace = egressNPNamespace.GetStringValue()
+	}
+	if egressNPRuleName, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyRuleName"); ok {
+		chFlowRow.egressNetworkPolicyRuleName = egressNPRuleName.GetStringValue()
+	}
+	if egressNPType, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyType"); ok {
+		chFlowRow.egressNetworkPolicyType = egressNPType.GetUnsigned8Value()
+	}
+	if egressNPRuleAction, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyRuleAction"); ok {
+		chFlowRow.egressNetworkPolicyRuleAction = egressNPRuleAction.GetUnsigned8Value()
+	}
+	if tcpState, _, ok := record.GetInfoElementWithValue("tcpState"); ok {
+		chFlowRow.tcpState = tcpState.GetStringValue()
+	}
+	if flowType, _, ok := record.GetInfoElementWithValue("flowType"); ok {
+		chFlowRow.flowType = flowType.GetUnsigned8Value()
+	}
+	if sourcePodLabels, _, ok := record.GetInfoElementWithValue("sourcePodLabels"); ok {
+		chFlowRow.sourcePodLabels = sourcePodLabels.GetStringValue()
+	}
+	if destinationPodLabels, _, ok := record.GetInfoElementWithValue("destinationPodLabels"); ok {
+		chFlowRow.destinationPodLabels = destinationPodLabels.GetStringValue()
+	}
+	if throughput, _, ok := record.GetInfoElementWithValue("throughput"); ok {
+		chFlowRow.throughput = throughput.GetUnsigned64Value()
+	}
+	if reverseThroughput, _, ok := record.GetInfoElementWithValue("reverseThroughput"); ok {
+		chFlowRow.reverseThroughput = reverseThroughput.GetUnsigned64Value()
+	}
+	if throughputFromSrcNode, _, ok := record.GetInfoElementWithValue("throughputFromSourceNode"); ok {
+		chFlowRow.throughputFromSourceNode = throughputFromSrcNode.GetUnsigned64Value()
+	}
+	if throughputFromDstNode, _, ok := record.GetInfoElementWithValue("throughputFromDestinationNode"); ok {
+		chFlowRow.throughputFromDestinationNode = throughputFromDstNode.GetUnsigned64Value()
+	}
+	if revTputFromSrcNode, _, ok := record.GetInfoElementWithValue("reverseThroughputFromSourceNode"); ok {
+		chFlowRow.reverseThroughputFromSourceNode = revTputFromSrcNode.GetUnsigned64Value()
+	}
+	if revTputFromDstNode, _, ok := record.GetInfoElementWithValue("reverseThroughputFromDestinationNode"); ok {
+		chFlowRow.reverseThroughputFromDestinationNode = revTputFromDstNode.GetUnsigned64Value()
+	}
+	return &chFlowRow
+}
+
+func (ch *ClickHouseExportProcess) flowRecordPeriodicCommit() {
+	commitTicker := time.NewTicker(commitInterval)
+	logTicker := time.NewTicker(time.Minute)
+	committedRec := 0
+	for {
+		select {
+		case <-ch.stopChan:
+			commitTicker.Stop()
+			logTicker.Stop()
+			return
+		case <-commitTicker.C:
+			committed, err := ch.batchCommitAll()
```

Comment:
it seems like ch.batchCommitAll() will return error when certain operations fail, but the error is not handled here?